pkg/framework/deployment.go

```diff
@@ -45,7 +56,13 @@ func DeleteDeployment(c client.Client, deployment *kappsapi.Deployment) error {
 
 // UpdateDeployment updates the specified deployment
 func UpdateDeployment(c client.Client, name, namespace string, updated *kappsapi.Deployment) error {
-	return wait.PollImmediate(RetryShort, WaitMedium, func() (bool, error) {
```

Comment:
why do we need to wrap this? now `UpdateDeployment ` is unused. Can't we just have a single `waitForDeploymentUpdate`?